### PR TITLE
doc: update function naming convention

### PR DESCRIPTION
**Pull request by shiqingg**
This patch updates coding guidelines about naming convention.

Signed-off-by: Shiqing Gao <shiqing.gao@intel.com>

### CODE REVIEW

**Comment by acrnsi** [comment]
OK to verify

File: doc/developer-guides/c_coding_guidelines.rst [hunk 1]
```diff
@@ -3501,7 +3501,13 @@ function. Some detailed rules are listed below:
    enough with prior rules, it shall be named with the module name as prefix, such
    as 'vie_read_mmio'.
 5) If the function is a wrapper of inline Assembly codes, it shall be named with
-   the format 'asm_<Assembly instruction mnemonic>', such as 'asm_pause'.
+   one of the following formats:
+
+   a) asm_<Assembly instruction mnemonic>, such as 'asm_pause'.
+   b) If the Assembly instruction mnemonic could not clearly indicate the
```
Comment (dbkinder) [inline doc/developer-guides/c_coding_guidelines.rst]:
```suggestion
   b) If the Assembly instruction mnemonic does not clearly indicate the
```

File: doc/developer-guides/c_coding_guidelines.rst [hunk 2]
```diff
@@ -3501,7 +3501,13 @@ function. Some detailed rules are listed below:
...
+      purpose of the function or the function includes multiple Assembly
+      instruction statements, the function shall be named with 'asm' as
```
Comment (dbkinder) [inline doc/developer-guides/c_coding_guidelines.rst]:
```suggestion
      instruction statements, the function shall be named with 'asm_' as
```

File: doc/developer-guides/c_coding_guidelines.rst [hunk 3]
```diff
@@ -3501,7 +3501,13 @@ function. Some detailed rules are listed below:
...
+      prefix and apply the prior rules.
```
Comment (dbkinder) [inline doc/developer-guides/c_coding_guidelines.rst]:
```suggestion
      prefix and apply the other non-assembly function naming rules.
```

File: doc/developer-guides/c_coding_guidelines.rst (hunk 2 above)
Comment (shiqingg) [reply doc/developer-guides/c_coding_guidelines.rst]: Hi @dbkinder,
Do you think we should update the following two rules as well?

From:
C-NC-03: Data structures exposed to external components shall be named with prefix ‘acrn’
C-NC-04: Data structures only used by hypervisor shall be named with prefix ‘hv’
To:
C-NC-03: Data structures exposed to external components shall be named with prefix ‘acrn_’
C-NC-04: Data structures only used by hypervisor shall be named with prefix ‘hv_’

**Comment by shiqingg** [comment]
@dbkinder Thanks for your suggestion! New version has been pushed.
Could you please help review it again and let me know your opinion about the inline comment?
Thanks!

File: doc/developer-guides/c_coding_guidelines.rst (hunk 2 above)
Comment (dbkinder) [reply doc/developer-guides/c_coding_guidelines.rst]: Yes, that would be a good change (to add the underscore), but be sure to escape the underscore so it doesn't look like a link:  'hv\_'

**Pull request merged**